controllers/movement.py

```diff
@@ -48,44 +63,74 @@ def setup(self):
         self.robot_object = self.field.getObject("auto_trajectory")
 
     def generate_trajectory(self) -> Trajectory:
+        # Grab all required information from self.chassis so we don't repeately run
+        # getters.
         self.chassis_velocity = self.chassis.get_velocity()
-
         self.x_pos = self.chassis.get_pose().X()
         self.y_pos = self.chassis.get_pose().Y()
-
         self.x_velocity = self.chassis_velocity.vx
         self.y_velocity = self.chassis_velocity.vy
 
-        self.trajectory_preprocess_pose = Pose2d(
-            self.x_pos, self.y_pos, math.atan2(self.y_velocity, self.x_velocity)
+        x_translation = self.goal.X() - self.x_pos
+        y_translation = self.goal.Y() - self.y_pos
```

Comment:
Instead of calculating these individually, do a vector subtraction of the Translation2ds. Something like
```python
translation = self.goal.translation() - pose.translation()
translation_distance = translation.norm()

Reply:
I can certainly do that. However both `x_translation` and `y_translation` are also used further in the code, shoud I still change the way `translation_distance` is calculated?